ec2utils/ec2utilsbase/lib/ec2utils/ec2preparer.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (c) 2018 SUSE LLC, Christian Bruckmayer <cbruckmayer@suse.com>
+#
+# This file is part of ec2utilsbase.
+#
+# ec2utilsbase is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ec2utilsbase is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ec2utilsbase.  If not, see <http://www.gnu.org/licenses/>.
+
+import boto3
+import os
+import random
+import datetime
+
+from ec2utils.ec2UtilsExceptions import EC2ConnectionException
+from ec2utils.ec2utils import EC2Utils
+from tempfile import mkstemp
+
+class EC2Preparer(EC2Utils):
+    """Class to prepare an Amazon EC2 account with all necessary resources"""
+
+    def __init__(self, access_key, account, key_pair_name, region, secret_key, security_group_ids, session_token, ssh_private_key_file, vpc_subnet_id, verbose):
+        EC2Utils.__init__(self)
+        self.access_key = access_key
+        self.account = account
+        self.key_pair_name = key_pair_name
+        self.region = region
+        self.secret_key = secret_key
+        self.session_token = session_token
+        self.ssh_private_key_file = ssh_private_key_file
+        self.vpc_subnet_id = vpc_subnet_id
+        self.verbose = verbose
+        self.security_group_ids = security_group_ids
+
+        self.ami = ''
+        self.temporary_key_created = False
+        self.temporary_vpc_created = False
+        self.temporary_security_group_created = False
+        self.vpc_id = ''
+        self.internet_gateway_id = ''
+        self.route_table_id = ''
+
+    # ---------------------------------------------------------------------
+    def clean_up(self):
+        if self.temporary_key_created:
+            self._remove_upload_key_pair()
+        if self.temporary_security_group_created:
+            self._remove_security_group()
+        if self.temporary_vpc_created:
+            self._remove_vpc()
+
+    # ---------------------------------------------------------------------
+    def _create_upload_key_pair(self, key_name='temporary_ec2_uploadkey'):
```

Comment:
Becomes a public method and should create it's own connection using ec2utils base class

Reply:
I'm already using a dedicated connection from ec2utils by using the _connect method, no?